yarn-project/ethereum/src/l1_tx_utils.ts

```diff
@@ -353,43 +405,73 @@ export class L1TxUtils {
     request: L1TxRequest,
     gasConfig?: Partial<L1TxUtilsConfig> & { fixedGas?: bigint; txTimeoutAt?: Date },
     blobInputs?: L1BlobInputs,
-  ): Promise<TransactionReceipt> {
-    const { txHash, gasLimit } = await this.sendTransaction(request, gasConfig, blobInputs);
-    return this.monitorTransaction(request, txHash, { gasLimit }, gasConfig, blobInputs);
+  ): Promise<{ receipt: TransactionReceipt; gasPrice: GasPrice }> {
+    const { txHash, gasLimit, gasPrice } = await this.sendTransaction(request, gasConfig, blobInputs);
+    const receipt = await this.monitorTransaction(request, txHash, { gasLimit }, gasConfig, blobInputs);
+    return { receipt, gasPrice };
   }
 
   /**
    * Gets the current gas price with bounds checking
    */
   private async getGasPrice(
     _gasConfig?: L1TxUtilsConfig,
+    isBlobTx: boolean = false,
     attempt: number = 0,
     previousGasPrice?: typeof attempt extends 0 ? never : GasPrice,
   ): Promise<GasPrice> {
     const gasConfig = { ...this.config, ..._gasConfig };
     const block = await this.publicClient.getBlock({ blockTag: 'latest' });
     const baseFee = block.baseFeePerGas ?? 0n;
 
-    // Get initial priority fee from the network
-    let priorityFee = await this.publicClient.estimateMaxPriorityFeePerGas();
+    // Get blob base fee if available
+    let blobBaseFee = 0n;
+    try {
+      const blobBaseFeeHex = await this.publicClient.request({ method: 'eth_blobBaseFee' });
+      blobBaseFee = BigInt(blobBaseFeeHex);
+      this.logger?.debug('Blob base fee:', { blobBaseFee: formatGwei(blobBaseFee) });
+    } catch {
+      this.logger?.warn('Failed to get blob base fee', attempt);
+    }
+
+    let priorityFee: bigint;
+    if (gasConfig.fixedPriorityFeePerGas) {
+      this.logger?.debug('Using fixed priority fee per gas', {
+        fixedPriorityFeePerGas: gasConfig,
+      });
+      // try to maintain precision up to 1000000 wei
+      priorityFee = BigInt(gasConfig.fixedPriorityFeePerGas * 1_000_000) * (WEI_CONST / 1_000_000n);
```

Comment:
Not sure I follow. Why not just `BigInt(gasConfig.fixedPriorityFeePerGas) * 1_000_000_000n`? Also, shouldn't `WEI_CONST` actually be `GWEI_CONST`?

Reply:
I think I was going for readability, to make it clear the 1,000,000 we're multiplying & dividing by is for precision.
As for naming I think either sort of works? happy to rename to `GWEI_CONST`

Reply:
I had initially thought that `WEI_CONST` would be 1e18. I checked ethers for inspiration and [they have `WeiPerEther`](https://docs.ethers.org/v5/api/utils/constants/#constants-WeiPerEther). Maybe `GWEI_IN_WEI`, as horrible as it looks?